src/components/Datefield/Datefield.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState } from "react";
+import Calendar from "react-calendar";
+import { Icon } from "../Icon/Icon";
+import { completeDateFilter, numbersAndSlashesFilter } from "../../utils";
+
+type IntrinsicElements = JSX.IntrinsicElements["input"];
+export interface Props extends IntrinsicElements {
+  id: string;
+  name: string;
+  label: string;
+  hint?: boolean;
+  disabled?: boolean;
+  defaultDate?: string;
+  minDate?: string;
+  maxDate?: string;
+  value?: string;
+}
+
+/**
+ * Datefield Component
+ * @param {string}  id                     A unique identifier for the input.
+ * @param {string}  name                   Name of the input field.
+ * @param {string}  label                  Field label.
+ * @param {boolean} [hint]                 Boolean that shows or hide the date format hint, in the format mm/dd/yyyy.
+ * @param {string}  [disabled]             Controls whether or not the date picker is disabled to the user.
+ * @param {string}  [defaultDate]          The date picker input will set this value if it is a valid date. The date should be in the format mm/dd/yyyy
+ * @param {string}  [minDate]              The date picker will not allow a date selection before this date. The date should be in the format mm/dd/yyyy
+ * @param {string}  [maxDate]              The date picker will not allow a date selection after this date. The date should be in the format mm/dd/yyyy.
+ * @param {string}  [value]                Value of the input element.
+ */
+
+export const Datefield: React.FC<Props> = ({
+  id,
+  name,
+  label,
+  value,
+  minDate,
+  maxDate,
+  defaultDate,
```

Comment:
Going to ask HCD on what todo when the user gives bad date data/ If we need to support any other date formats